Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -130,8 +130,9 @@ final class PurchasesOrchestrator {
         storeKit2TransactionListener.delegate = self
         storeKit2StorefrontListener.delegate = self
 
+        storeKit2TransactionListener.listenForTransactions()
```

Comment:
Is there any danger of this sending the same transaction multiple times and/or with different data to our backend?

Reply:
The only danger is increased load. @aboedo and I talked about this yesterday.
#2590 adds a test that verifies this particular scenario and checks that we only post once.

Reply:
Yeah, definitely a compromise. Our system is pretty good at de-duping post receipts that are in flight, but there's always a chance that we get slightly delayed signals from SK1 and SK2 and the response from one gets back before we post the other one and we do a double post. 

But on the other hand, that seems like a better scenario than if we miss a purchase in observer mode because we were observing the wrong object.